### PR TITLE
Clean up Makefile

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description
A few sentences describing the overall goals of the pull request's commits.
Please include
- the type of fix - (e.g. bug fix, new feature, documentation)
- some details on _why_ this PR should be merged
- the details of the testing you've done on it (both manual and automated)
- which components are affected by this PR
- links to issues that this PR addresses

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,27 +1,87 @@
...
-SRCDIR=.
+# Makefile configuration options 
 CONTAINER_NAME=calico/kube-policy-controller
-
-default: all
-all: policy-controller
-
-# Build the calico/kube-policy-controller Docker container.
+PACKAGE_NAME?=github.com/projectcalico/k8s-policy
+GO_BUILD_VER:=latest
```
Comment (bcreane) [inline Makefile]: pinned to 0.7 elsewhere ... this looks fine to me, just wanted to call it out.

File: Makefile [hunk 2]
```diff
@@ -1,27 +1,87 @@
...
+# Build targets 
+###############################################################################
+## Builds the docker image
 docker-image: image.created
+image.created: dist/kube-policy-controller
+	# Build the docker image for the policy controller.
+	docker build -t $(CONTAINER_NAME) .
+	touch image.created
 
-# Run the unit tests.
```
Comment (bcreane) [inline Makefile]: UTs are still in-progress?

File: Makefile [hunk 3]
```diff
@@ -1,27 +1,87 @@
...
 CONTAINER_NAME=calico/kube-policy-controller
-
-default: all
-all: policy-controller
-
-# Build the calico/kube-policy-controller Docker container.
+PACKAGE_NAME?=github.com/projectcalico/k8s-policy
+GO_BUILD_VER:=latest
+CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)
+LDFLAGS=-ldflags "-X $(PACKAGE_NAME)/calicoctl/commands.VERSION=$(CALICOCTL_VERSION) \
```
Comment (bcreane) [inline Makefile]: is this copy/paste from calicoctl/Makefile? If so, don't you need to define $CALICOCTL_VERSION, $CALICOCTL_BUILD_DATE and $CALICOCTL_GIT_REVISION? At the least, you'll want to make sure they're properly initialized.

File: Makefile [hunk 4]
```diff
@@ -1,27 +1,87 @@
...
+	$(MAKE) OS=linux ARCH=amd64 binary-containerized
+	mv dist/kube-policy-controller-linux-amd64 dist/kube-policy-controller
+
+## Populates the vendor directory.
+vendor: glide.yaml
+	# Ensure that the glide cache directory exists.
+	mkdir -p $(HOME)/.glide
+
+	# To build without Docker just run "glide install -strip-vendor"
+	if [ "$(LIBCALICOGO_PATH)" != "none" ]; then \
```
Comment (bcreane) [inline Makefile]: shouldn't LIBCALICOGO_PATH have a default initialization somewhere? `LIBCALICOGO_PATH?=none`

File: Makefile [hunk 5]
```diff
@@ -1,27 +1,87 @@
...
+###############################################################################
+# Test targets 
+###############################################################################
+
+## Runs all tests - good for CI. 
+ci: clean docker-image ut st
 
-# Run system tests.
+GET_CONTAINER_IP := docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}'
+K8S_VERSION=1.6.4
```
Comment (bcreane) [inline Makefile]: something more recent, 1.7.4?

**Comment by caseydavenport** [comment]
CC @svInfra17

File: Makefile (hunk 2 above)
Comment (caseydavenport) [reply Makefile]: Yep, these will be in a separate PR.

**Comment by caseydavenport** [comment]
@bcreane thanks, I've updated according to your comments.

**Pull request merged**